src/System.Text.Json/tests/WritableJsonApiTests.JsonNumberUnitTests.cs

```diff
@@ -0,0 +1,218 @@
+﻿// Licensed to the .NET Foundation under one or more agreements.
+// The .NET Foundation licenses this file to you under the MIT license.
+// See the LICENSE file in the project root for more information.
+
+using Xunit;
+
+namespace System.Text.Json
+{
+//#pragma warning disable xUnit1000
+    public static partial class WritableJsonApiTests
+//#pragma warning restore xUnit1000
+    {
+        [Theory]
+        [InlineData(-456)]
+        [InlineData(0)]
+        [InlineData(17)]
+        [InlineData(Int64.MaxValue)]
+        [InlineData(2.3)]
+        [InlineData(-17.009)]
+        [InlineData(3.14f)]
+        [InlineData(0x2A)]
+        [InlineData(0b_0110_1010)]
+        [InlineData("1e400")]
+        [InlineData("1e+100000002")]
+        [InlineData("184467440737095516150.184467440737095516150")]
+        public static void TestJsonNumberConstructor(object value)
+        {
+            // should not throw any exceptions:
+            var jsonNumber = new JsonNumber(value as dynamic);
+        }
+
+        [Fact]
+        public static void TestByte()
+        {
+            var jsonNumber = new JsonNumber();
+            byte value = 17;
+            
+            jsonNumber.SetByte(value);
+            byte result = jsonNumber.GetByte();
+            Assert.Equal(value, result);
+            
+            bool success = jsonNumber.TryGetByte(out result);
+            Assert.True(success);
+            Assert.Equal(value, result);
+        }
+
+        [Fact]
+        public static void TestShort()
+        {
+            var jsonNumber = new JsonNumber();
```

Comment:
The fact that this type has a default ctor surprised me.  There should be a test that tests the behaviors of this ctor (what throws, what succeeds, etc).

Either it should be 0 (in which case all methods work) or it should not exist.

Reply:
Yep, same here.

Reply:
Ok, I can change it to "0".